### PR TITLE
fix(dora): add deployment generator

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. In some plugins, such as Jenkins, user can set a job without any commit, so when converting cicd_pipelines to cicd_deployments, if pipeline has no commits, this progress should work well.
This PR just dispart cicd_deployments generator and make sure cicd_deployments can be generated anyway.
2. I have checked all plugins that can generate cicd_deloyments table,and make sure that if there is no commit associated to its original builds,  devlake can generate cicd_deployments table anyway.

### Does this close any open issues?
Closes N/A

### Screenshots
Include any relevant screenshots here.
image
image
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/dora/impl/impl.go [hunk 1]
```diff
@@ -130,6 +131,7 @@ func (p Dora) MakeMetricPluginPipelinePlanV200(projectName string, options json.
 					"projectName": projectName,
 				},
 				Subtasks: []string{
+					tasks.DeploymentGeneratorMeta.Name,
```
Comment (abeizn) [inline backend/plugins/dora/impl/impl.go:134]: Please unify it, either use xxx.Name or write it out directly.

Comment (d4x1) [reply backend/plugins/dora/impl/impl.go]: Updated.

**Review by abeizn — APPROVED** [review]
lgtm

**Comment by d4x1** [comment]
@Startrekzky  I think some sql scripts in dashboard shoule be updated too.
Such as 
image

**Pull request merged**